### DISCUSSION TITLE
Turbomachinery adjoint objectives and  Total-Total Eff computation

### DISCUSSION THREAD

**Discussion by wkangholy** [su2code/SU2#1630] [Q&A]
**Describe the bug**
1.
*A clear and concise description of what the bug is and what you expect the behavior to be instead. If applicable, add screenshots to help explain your problem.*

When I tried to run the TestCases/disc_adj_turbomachinery/transonic_stator_2D, it showed the error.
Seem the error was reported in https://github.com/su2code/SU2/issues/889.
But in version 7.3.1, it still has similar issues.
May I ask how to use updateHistoryMap.py to add custom objectives? Thanks.

raceback (most recent call last):
  File "/usr/local/bin/shape_optimization.py", line 183, in <module>
    main()
  File "/usr/local/bin/shape_optimization.py", line 92, in main
    shape_optimization( options.filename    ,
  File "/usr/local/bin/shape_optimization.py", line 155, in shape_optimization
    project = SU2.opt.Project(config,state)
  File "/usr/local/bin/SU2/opt/project.py", line 129, in __init__
    group = historyOutFields[this_obj]['GROUP']
KeyError: 'ENTROPY_GENERATION'

2
When I run the case in TestCases/turbomachinery/axial_stage_2D, the value for  Total-Total Eff is increased with the total iterative step.

When I set Number of total iterations = 3000, the EFF is increased than Number of total iterations =1000. But the pressure ratio remains unchanged.

**To Reproduce**
*Please provide a config file and mesh for a minimal example that recreates the problem.*

 TestCases/turbomachinery/axial_stage_2D

**Desktop (please complete the following information):**
 - OS: [e.g., Linux (Ubuntu 20.04)]
 - C++ compiler and version: [e.g., g++ (GCC) 9.4.0]
 - MPI implementation and version: [e.g., OpenMPI 4.0.3]
 - SU2 Version: [e.g., v7.3.1]

**Comment by pcarruscag** [comment]
For 1) the issue is that entropy generation is an objective but not an output, and so it is not possible to map it, and so the python scripts do not know how to obtain the value of the function.
This is being worked in #1316, until then you may consider using FADO https://su2code.github.io/tutorials/Species_Transport/

For 2) I'm converting this to a discussion.

**Comment by wkangholy** [comment]
Thanks for the quick response. I will try the python script instead of shape_optimization.py.
